corgi/core/models.py

```diff
@@ -495,21 +500,16 @@ def manifest(self) -> str:
 
     def get_latest_components(self):
         """Return root components from latest builds."""
-        root_components = (
-            Q(type=Component.Type.RPM, arch="src")
-            | Q(type=Component.Type.CONTAINER_IMAGE, arch="noarch")
-            | Q(type=Component.Type.RPMMOD)
-        )
         return (
             Component.objects.filter(
-                root_components,
+                ROOT_COMPONENTS_CONDITION,
                 product_streams__overlap=(self.ofuri,),
             )
             .annotate(
-                latest=Subquery(
+                latest=models.Subquery(
                     Component.objects.filter(
-                        root_components,
-                        name=OuterRef("name"),
+                        ROOT_COMPONENTS_CONDITION,
```

Comment:
nit: I wonder if we can also level the ComponentNode(level=0) trick here as ComponentTypes at the root level are likely to change in future.

Reply:
We could! I wasn't sure about the logic for ROOT_COMPONENTS, but I can definitely switch it to report only components with level=0 / at the root of a tree.

Reply:
Actually, we could do this but it would require some more invasive changes. Right now we filter on Component, so we can't check for level=0. We'd need to switch to filtering ComponentNodes with level=0, then look up (and dedupe) their linked Component objects. So I'll leave this for another MR.